editor/src/core/shared/array-utils.ts

```diff
@@ -433,3 +433,7 @@ export function strictEvery<T>(
 ): boolean {
   return ts.length > 0 && ts.every(predicate)
 }
+
+export function isNotNull<T>(value: T | null | undefined): value is T {
```

Comment:
Why is this in `array-utils.ts`?

Reply:
This is the only use case of isNotNull that is important:

``` typescript
const exampleArray = [1, 2, null, 4, 5] // this is type Array<number | null>

const filteredArrayBad = exampleArray.filter(e => e != null) // this is type Array<number | null>

const filteredArrayGood = exampleArray.filter(isNotNull) // this is type Array<number> !! 
```

I couldn't really find a better util file for this, so I put it in the array utils

Reply:
yeah that might work! :)

Reply:
Huh, I could have sworn we already had something like this for exactly this use case, but I can't find it...